### PR TITLE
Add llm-benchmarks proposal

### PR DESCRIPTION
**Pull request by IcyFeather233**
**What type of PR is this?**

OSPP proposal

**What this PR does / why we need it**:

Investigated various aspects of implementing the LLM Benchmark in Ianvs, introduced the integration plan of opencompass, and achieved the dataset map.

### CODE REVIEW

**Comment by IcyFeather233** [comment]
By the way, my implementation is in https://github.com/IcyFeather233/ianvs/tree/dev, you can follow [this readme](https://github.com/IcyFeather233/ianvs/tree/dev/examples/llm/singletask_learning_bench/simple_qa) to try llm single task learning and opencompass

**Comment by hsj576** [comment]
The PR should be translated into English to facilitate understanding by people from other countries.

**Comment by IcyFeather233** [comment]
> https://github.com/IcyFeather233/ianvs/tree/dev

Thanks for the advice! I will translate it after the proposal content will not be changed.

File: docs/proposals/scenarios/llm-benchmarks/llm-benchmarks.md
Comment (MooreZheng) [inline docs/proposals/scenarios/llm-benchmarks/llm-benchmarks.md:1]: Great to see a comprehensive proposal. This one is close to the final version. As discussed in the routine meeting, there might be some advices. 
1. Though the directories are clear, an architecture is still needed to clarify what is modified in this project. E.g., what is in the TestEnv, TestCase. It seems to me that at least we have made changes on the data format to support NLP. You might want to take a look at https://github.com/kubeedge/ianvs/pull/122/files for an architecture example.  
2. Prompts would have different forms, e.g., N shots. How shall we develop a prompt template to adapt all these? It looks interesting and challenging.

**Comment by IcyFeather233** [comment]
@MooreZheng Now I add my changes to ianvs core, including 2 flowcharts showing the architecture. And a more complete benchmark format is updated in the doc.

File: docs/proposals/scenarios/llm-benchmarks/llm-benchmarks.md
Comment (IcyFeather233) [reply docs/proposals/scenarios/llm-benchmarks/llm-benchmarks.md:1]: Now I add my changes to ianvs core, including 2 flowcharts showing the architecture. And a more complete benchmark format is updated in the doc.

**Review by MooreZheng — COMMENTED** [review]
Overall looks good to me. Just a few comments would be considered: 
1. The revision for index in env manager of ianvs core, which should be highlighted: The proposed design also aims to support both the old and new index.  
2.  The integration part of OpenCompass should also be highlighted in the structure.png. A tutorial is needed.  
3.  The prompt template is extendable, including the keys and prompts, which can be also highlighted in the proposal.

**Review by MooreZheng — APPROVED** [review]
/lgtm

**Comment by ojuschugh1** [comment]
Hi @IcyFeather233 , I wanted to discuss one thing regarding this project. Is there any way I can chat/discuss this with you?

**Comment by MooreZheng** [comment]
> Hi @IcyFeather233 , @hsj576 , @MooreZheng , I wanted to discuss one thing regarding this project. Is there any way I can chat/discuss this with you? I am unable to find anyone of you on slack channel

Hi, the slack channel currently has networking issues for Chinese users. You might want to leave your comment here or raise a issue.

**Comment by MooreZheng** [comment]
Related issue and PR:  
Phase 1 Gov LLM https://github.com/kubeedge/ianvs/issues/95 https://github.com/kubeedge/ianvs/pull/113 https://github.com/kubeedge/ianvs/pull/144
Phase 2 Gov RAG https://github.com/kubeedge/ianvs/issues/177 https://github.com/kubeedge/ianvs/pull/186 https://github.com/kubeedge/ianvs/pull/196

**Pull request merged**